test/state/page-config/actions.test.js

```diff
@@ -204,7 +230,116 @@ describe('state/page-config/actions', () => {
         expect(actionTypes).toEqual([SET_PAGE_WIDGET]);
         expect(putPageWidget).toHaveBeenCalledWith(CURRENT_PAGE_CODE, FRAME_ID, WIDGET);
         done();
-      });
+      }).catch(done.fail);
+    });
+  });
+
+  describe('setSelectedPageOnTheFly()', () => {
+    beforeEach(() => {
+      store = mockStore(INITIAL_STATE);
+    });
+
+    it('dispatches REMOVE_PAGE_WIDGET if value = false', (done) => {
+      getSelectedPageModelMainFrame.mockReturnValue({ pos: 3 });
+      store.dispatch(setSelectedPageOnTheFly(false)).then(() => {
+        const actionTypes = store.getActions().map(action => action.type);
+        expect(actionTypes).toEqual([REMOVE_PAGE_WIDGET]);
```

Comment:
I understand why you are doing this, but I don't know if computationally is any better than just checking the array manually

Reply:
Not computationally, but it's a way to easily read the sequence of actions being dispatched.
Do you think it would be better to check manually?

Reply:
it's ok either way really :stuck_out_tongue: